### PR TITLE
Refactor all inputs

### PR DESCRIPTION
**Pull request by clayton-duarte**
Refactored the following input components  to use the same base Input:
- Input
- InputNumber
- TextField
- InputAutocomplete

### CODE REVIEW

File: src/components/InputAutocomplete/index.spec.tsx [hunk 1]
```diff
@@ -5,7 +5,7 @@ import ThemeProvider from '../ThemeProvider'
 import InputAutocomplete, { InputAutoCompleteProps } from '.'
 
 describe('InputAutocomplete', () => {
-  const mockOnChange = jest.fn((a) => a)
+  const mockOnChange = jest.fn((a) => a) as InputAutoCompleteProps['onChange']
```
Comment (l0rdcafe) [inline src/components/InputAutocomplete/index.spec.tsx:8]: question: do we need this cast?

Comment (clayton-duarte) [reply src/components/InputAutocomplete/index.spec.tsx:8]: actually dont, hehe, lemme remove

File: src/components/Label/__snapshots__/index.spec.tsx.snap [hunk 2]
```diff
@@ -13,3 +13,17 @@ exports[`Label should render properly 1`] = `
...
+  pointer-events: none;
+  font-size: 14px;
+}
+
+<label
+  class="emotion-0"
+>
+  label
+</label>
+`;
```
Comment (l0rdcafe) [inline src/components/Label/__snapshots__/index.spec.tsx.snap:29]: thanks for the test

File: src/components/TextField/index.tsx [hunk 3]
```diff
@@ -1,67 +1,36 @@
...
+  placeholder, // yes, we are filtering the placeholder and not sending it to the input element on purpose
   children,
-  onChange,
-  onBlur,
   hasError,
-  onFocus,
   label,
-  name,
-  ...gridProps
+  ...inputProps // HTMLAttributes<HTMLInputElement>
```
Comment (l0rdcafe) [inline src/components/TextField/index.tsx:26]: nicely done

**Review by l0rdcafe — DISMISSED** [review]

File: src/components/Label/__snapshots__/index.spec.tsx.snap (hunk 2 above)
Comment (clayton-duarte) [reply src/components/Label/__snapshots__/index.spec.tsx.snap:29]: hehe, I was simplifying the component and realized that it was missing the only test case possible

File: src/components/TextField/index.tsx (hunk 3 above)
Comment (clayton-duarte) [reply src/components/TextField/index.tsx:26]: hehe, a little better now, we dont need access to the wrapper, but to the input

**Review by l0rdcafe — APPROVED** [review]

**Pull request merged**